### DISCUSSION TITLE
Automatically remove home folder when user are deleted

### DISCUSSION THREAD

**Discussion by Dante3004** [drakkan/sftpgo#698] [Q&A] [answered]
Hello, when we delete a user, his home folder is not deleted on the server and it is not proposed from the web interface. Is there a setting I missed that allows me to choose whether or not to keep the user's home folder?

Thanks in advance for your answers :)

**Comment by drakkan** [comment, marked as answer]
Hi,

this is not supported and very difficult to implement for filesystem different from the local one. For local filesystem it is dangerous since you can easily delete unwanted files, think for example if you have two users with the same home dir or similar.

However you can easily customize SFTPGo to add this feature yourself. You can configure a custom action to react to the `delete` provider event and remove the home dir in your custom script/http hook.

Take a look [here](https://github.com/drakkan/sftpgo/blob/main/docs/custom-actions.md#provider-events) for more details.

**Comment by Dante3004** [comment]
It is indeed for local and we understand the risks that this can bring.I thank you for your answer, I will follow your advice and create a personalized action
